### ISSUE TITLE
[Bug]: Message content can not be null or empty even though it is allowed by Discord API

### ISSUE THREAD

**Issue by Blu3SoulsIT** [RogueException/Discord.Net#2689]
### Check The Docs

- [X] I double checked the docs and couldn't find any useful information.

### Verify Issue Source

- [X] I verified the issue was caused by Discord.Net.

### Check your intents

- [X] I double checked that I have the required intents.

### Description

Assume you just want to send a message component and nothing else. You would write something like
```csharp
ISocketMessageChannel channel = ...;
await channel.SendMessageAsync(string.Empty, components: components);
```
This should work as it is allowed to omit message content when you send a component as described [here](https://discord.com/developers/docs/resources/channel#create-message):
> \* At least one of content, embeds, sticker_ids, components, or files[n] is required.

However, it does not work since the message content gets checked against being null or empty in the DiscordRestApiClient.CreateMessageAsync method even if the message has components.

### Version

3.10.0

### Logs

Important part of the stack trace.
```
System.ArgumentException: Argument cannot be blank. (Parameter 'Content')
   at Discord.Preconditions.NotNullOrEmpty(String obj, String name, String msg)
   at Discord.API.DiscordRestApiClient.CreateMessageAsync(UInt64 channelId, CreateMessageParams args, RequestOptions options)
   at Discord.Rest.ChannelHelper.SendMessageAsync(IMessageChannel channel, BaseDiscordClient client, String text, Boolean isTTS, Embed embed, AllowedMentions allowedMentions, MessageReference messageReference, MessageComponent components, ISticker[] stickers, RequestOptions options, Embed[] embeds, MessageFlags flags)
```

### Sample

```csharp
// Create a valid config.
DiscordSocketConfig config = null!;

var client = new DiscordSocketClient(config);

// Set channel id to any message channel.
ulong channelId = default;
if (await client.GetChannelAsync(channelId) is ISocketMessageChannel channel)
{
    // Create any component.
    var components = new ComponentBuilder()
        .WithButton(new ButtonBuilder()
            .WithCustomId("ButtonId"))
        .Build();

    // Try to send message with component and empty text.
    // This is where the error happens.
    await channel.SendMessageAsync(string.Empty, components: components);
}
```

### Packages

`Discord.Addons.Hosting 5.2.0` (This should not change anything)

**Comment by Misha-133** [comment]
This was fixed in #2609 & will come in next release
As for now you can do one of following:
- use markdown "bug" => set the content to `_ _`, as this will result in visually empty message content
- build the library yourself from source 
- use [MyGet nightly builds](https://github.com/discord-net/Discord.Net#unstable) (Just note that `Discord.Net` package on myget has bugged deps, so you'd need to install all `Discord.Net.X` packages one by one (`X` being `Core`, `Rest`, `WebSocket`, `Interactions`, `Commands`, `Webhook`))

**Comment by Blu3SoulsIT** [comment]
Thank you for the information.
